# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1502373335 14400
#      Thu Aug 10 09:55:35 2017 -0400
# Node ID 3c5a49805dbdfa39c32a58583431a59f2078deab
# Parent  7148b237f94f0944dcd8ddc965f14536a0c5b13c
get default pager from octave::build_env, not macro (bug #51718)

* pager.cc (default_pager): Use octave::build_env::DEFAULT_PAGER, not
the obsolete OCTAVE_DEFAULT_PAGER configuration macro.

diff --git a/libinterp/corefcn/pager.cc b/libinterp/corefcn/pager.cc
--- a/libinterp/corefcn/pager.cc
+++ b/libinterp/corefcn/pager.cc
@@ -28,16 +28,17 @@ along with Octave; see the file COPYING.
 #include <iostream>
 #include <string>
 
 #include "cmd-edit.h"
 #include "oct-env.h"
 #include "oct-syscalls.h"
 #include "singleton-cleanup.h"
 
+#include "build-env.h"
 #include "defaults.h"
 #include "defun.h"
 #include "error.h"
 #include "errwarn.h"
 #include "input.h"
 #include "octave.h"
 #include "ovl.h"
 #include "pager.h"
@@ -59,20 +60,18 @@ static std::string diary_file ("diary");
 // The diary file.
 static std::ofstream external_diary_file;
 
 static std::string
 default_pager (void)
 {
   std::string pager_binary = octave::sys::env::getenv ("PAGER");
 
-#if defined (OCTAVE_DEFAULT_PAGER)
   if (pager_binary.empty ())
-    pager_binary = OCTAVE_DEFAULT_PAGER;
-#endif
+    pager_binary = octave::build_env::DEFAULT_PAGER;
 
   return pager_binary;
 }
 
 // The shell command to run as the pager.
 static std::string VPAGER = default_pager ();
 
 // The options to pass to the pager.
